### PR TITLE
Update to Stripe v3

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Stripe now required to use the latest version of Stripe checkout (elements) for auto compliance, this updates the integration to use Stripe elements (also known as Stripe checkout v3)

# Impacts
Stripe integration

# Delete branch after merge
YES

# NEWS file entries
## Enhancements
* Updated Stripe integration to use Stripe Elements

### CODE REVIEW

**Comment by julsemaan** [comment]
There is 1 last detail to figure out, I had to go back to creating one customer entry per subscription (instead of reusing the customer entry) due to how declines are now handled.

Reached out to Stripe regarding that, we'll see if I have a way to put it back to using the same customer object, otherwise we'll merge as-is.

**Comment by julsemaan** [comment]
@jrouzierinverse, please review this for inclusion in 10.3

File: lib/pf/Authentication/Source/StripeSource.pm [hunk 1]
```diff
@@ -240,12 +252,15 @@ sub subscribe_customer {
...
 
     my ($code, $data);
-    if(my $cus = $self->get_customer_by_email($session->{email})) {
-        ($code, $data) = $self->_send_form($self->curl, "v1/customers/".$cus->{id}, $object);
-    }
-    else {
-        ($code, $data) = $self->_send_form($self->curl, "v1/customers", $object);
+    ($code, $data) = $self->_send_form($self->curl, "v1/customers", $object);
+
+    my $subscriptions = $self->_get_json($self->curl, "v1/subscriptions?customer=".$data->{id});
```
Comment (jrouzierinverse) [inline lib/pf/Authentication/Source/StripeSource.pm]: ```$data->{id}``` needs to url encoded

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

File: lib/pf/Authentication/Source/StripeSource.pm (hunk 1 above)
Comment (julsemaan) [reply lib/pf/Authentication/Source/StripeSource.pm]: Done

**Pull request merged**